pkg/analysis_server/lib/src/services/correction/fix_internal.dart

```diff
@@ -1875,6 +1884,25 @@ class FixProcessor {
     }
   }
 
+  void _addFix_removeTypeName() {
+    final FunctionTypedFormalParameter functionTyped =
+        node is FunctionTypedFormalParameter
+            ? node
+            : node.getAncestor((node) => node is FunctionTypedFormalParameter);
+    if (functionTyped != null) {
+      _addReplaceEdit(rf.rangeNode(functionTyped),
+          utils.getNodeText(functionTyped.identifier));
+      _addFix(DartFixKind.REMOVE_TYPE, []);
+      return;
+    }
```

Comment:
Yes you were right, the behavior always will prefer to remove the function-typed formal parameter and this is correct for our rules, for example if you have this:

```dart
void callFunction(/*LINT*/void f(dynamic p)) {}
```

The quick-fix (without considering the meaning of the lints) is ambiguous, it could refer the typeName of the returnType (void in this case) or the function-typed formal parameter, and the fix always prefer the function-typed formal parameter and that is because is the behavior needed for fix the lint rules now, so the solution is correct but...

The problem could appear if someone would like to use this fix in the future, so it is relevant to know what you noticed to take a decision, so I added documentation to the quick-fix explaining this behavior.